examples/lm1b/train.py

```diff
@@ -542,7 +572,7 @@ def encode_strings(strs, max_len):
       if step % config.eval_every_steps == 0 or is_last_step:
         with report_progress.timed("training_metrics"):
           logging.info("Gathering training metrics.")
-          train_metrics = common_utils.get_metrics(train_metrics)
+          train_metrics = common_utils.stack_forest(train_metrics)
```

Comment:
why was this necessary?

Reply:
[`get_metrics`](https://github.com/google/flax/blob/main/flax/training/common_utils.py#L72) is a utility function for pmap to get a single copy of a device-replicated metric. It calls `stack_forest` on the metrics.
If I try using `get_metrics` with `jit`, I get:
```
File "/home/marcuschiam/.local/lib/python3.8/site-packages/flax/training/common_utils.py", line 83, in <lambda>
    device_metrics = jax.tree_util.tree_map(lambda x: x[0], device_metrics)
  File "/home/marcuschiam/.local/lib/python3.8/site-packages/jax/_src/array.py", line 317, in __getitem__
    return lax_numpy._rewriting_take(self, idx)
  File "/home/marcuschiam/.local/lib/python3.8/site-packages/jax/_src/numpy/lax_numpy.py", line 4143, in _rewriting_take
    return _gather(arr, treedef, static_idx, dynamic_idx, indices_are_sorted,
  File "/home/marcuschiam/.local/lib/python3.8/site-packages/jax/_src/numpy/lax_numpy.py", line 4152, in _gather
    indexer = _index_to_gather(shape(arr), idx)  # shared with _scatter_update
  File "/home/marcuschiam/.local/lib/python3.8/site-packages/jax/_src/numpy/lax_numpy.py", line 4253, in _index_to_gather
    idx = _canonicalize_tuple_index(len(x_shape), idx)
  File "/home/marcuschiam/.local/lib/python3.8/site-packages/jax/_src/numpy/lax_numpy.py", line 4586, in _canonicalize_tuple_index
    raise IndexError(
IndexError: Too many indices for array: 1 non-None/Ellipsis indices for dim 0.
```

Reply:
Cool, thanks!